src/projection/shared/stack.ts

```diff
@@ -106,4 +107,14 @@ export class NodeStack {
             node.instance && node.scheduleRender(false)
         })
     }
+
+    /**
+     * Clear any leads that have been removed this render to prevent them from being
+     * used in future animations and to prevent memory leaks
+     */
+    clearLeadSnapshot() {
```

Comment:
Not sure if "clear" is used in other places too, but something like `removeLeadSnapshot` makes a bit more sense to me.